### PR TITLE
Cospad QMK Configurator support and refactor

### PR DESCRIPTION
**Pull request by mechmerlin**
- Changed `KEYMAP` to `LAYOUT_ortho`
- Added a new LAYOUT macro called `LAYOUT_numpad`
- Changed keymap.c to reflect new changes
- Added info.json for Configurator support

### CODE REVIEW

File: keyboards/cospad/cospad.h [hunk 1]
```diff
@@ -56,12 +56,25 @@
 	{k40, k41, k42, k43}, \
 	{k50, k51, k52, k53} \
 }
+
+#define LAYOUT_numpad( \
```
[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/cospad/cospad.h]: I'm not sure I can get the matrix to conform to that without a hardware change. Those are the switch positions specified by the manufacturer.

File: keyboards/cospad/cospad.h [hunk 2]
```diff
@@ -40,8 +40,8 @@
  */
 // The first section contains all of the arguments
 // The second converts the arguments into a two-dimensional array
-#define KEYMAP( \
-    	k00, k01, k02, k03, \
+#define LAYOUT_ortho( \
```
[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/cospad/cospad.h]: done

File: keyboards/cospad/cospad.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/cospad/cospad.h]: The physical matrix no, but that's the entire point of the layout macro, you can change it however you want, and it doesn't affect the electrical matrix. 

This should work: 

```c
#define LAYOUT_numpad_6x4( \
  k00, k01, k02, k03, \
  k10, k11, k12, k13, \
  k20, k21, k22, \
  k30, k31, k32,  k23,   \
  k40, k41, k42,  \
  k50,      k52,      k43       \
) \
{ \
    {k00, k01,   k02, k03}, \
    {k10, k11,   k12, k13}, \
    {k20, k21,   k22, k23}, \
    {k30, k31,   k32, KC_NO}, \
    {k40, k41,   k42, k43}, \
    {k50, KC_NO, k52, KC_NO} \
}
```

[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/cospad/cospad.h]: Ohh I see what you mean, I thought you wanted me to change the numbers not the positioning lol. I did the same thing with the alf_x2 not too long ago.

[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/cospad/cospad.h]: Sorry that I wasn't clear enough! 

But yeah, I mean the "top" part.  As long as the variables match, then rearranging them  will work just fine.

**Comment by drashna** [comment]
Looks like you missed the layout name changes in the cospad.h file.

**Comment by drashna** [comment]
Thanks!

**Comment by noroadsleft** [comment]
@drashna I was just thinking a day or two ago about refactoring the five- and six-row numpad layouts to move the Plus and Enter keys up a row, because I think that's more intuitive to users. Nothing says the physical and electrical locations have to be on the same rows in both parts of the `#define`.

```
#define LAYOUT_numpad_6x4( \
    k00, k01, k02, k03, \
    k10, k11, k12, k13, \
    k20, k21, k22, k23, \
    k30, k31, k32,      \
    k40, k41, k42, k43, \
    k50,      k52       \
) \
{ \
    { k00, k01,   k02, k03   }, \
    { k10, k11,   k12, k13   }, \
    { k20, k21,   k22, k23   }, \
    { k30, k31,   k32, KC_NO }, \
    { k40, k41,   k42, k43   }, \
    { k50, KC_NO, k52, KC_NO }  \
}
```

The top half could stay as it's shown above, but the corresponding parameters in the two-dimensional array could move as needed, depending on how the matrix is wired. I think that might make more sense than how we have it now.

**Comment by drashna** [comment]
Either would work, but it definitely needs to be consistent between all of them.  
And since we already had the layout... 

But, the problem is that I can see both being "correct".  So, let's pick a format and stick with it.

**Pull request merged**